native/src/main/java/io/ballerina/lib/ibm.ibmmq/utils/CommonUtils.java

```diff
@@ -0,0 +1,202 @@
+        } else if (value instanceof Byte) {
+            mqMessage.setByteProperty(key.getValue(), propertyDescriptor, (Byte) properties.get(key));
+        } else if (value instanceof byte[]) {
+            mqMessage.setBytesProperty(key.getValue(), propertyDescriptor, ((byte[]) properties.get(key)));
+        } else if (value instanceof Float) {
+            mqMessage.setFloatProperty(key.getValue(), propertyDescriptor, (Float) properties.get(key));
+        } else if (value instanceof Double) {
+            mqMessage.setDoubleProperty(key.getValue(), propertyDescriptor, (Double) properties.get(key));
+        } else if (value instanceof BString) {
+            mqMessage.setStringProperty(key.getValue(), propertyDescriptor, ((BString) properties.get(key)).getValue());
+        }
+    }
+
+    private static MQPropertyDescriptor getMQPropertyDescriptor(BMap descriptor) {
+        MQPropertyDescriptor propertyDescriptor = new MQPropertyDescriptor();
+        if (descriptor.containsKey(PD_VERSION)) {
+            propertyDescriptor.version = ((Long) descriptor.get(PD_VERSION)).intValue();
+        }
+        if (descriptor.containsKey(PD_COPY_OPTIONS)) {
+            propertyDescriptor.copyOptions = ((Long) descriptor.get(PD_COPY_OPTIONS)).intValue();
+        }
+        if (descriptor.containsKey(PD_OPTIONS)) {
+            propertyDescriptor.options = ((Long) descriptor.get(PD_OPTIONS)).intValue();
+        }
+        if (descriptor.containsKey(PD_SUPPORT)) {
+            propertyDescriptor.support = ((Long) descriptor.get(PD_SUPPORT)).intValue();
+        }
+        if (descriptor.containsKey(PD_CONTEXT)) {
+            propertyDescriptor.context = ((Long) descriptor.get(PD_CONTEXT)).intValue();
+        }
+        return propertyDescriptor;
+    }
+
+    private static BMap populateDescriptorFromMQPropertyDescriptor(MQPropertyDescriptor propertyDescriptor) {
+        BMap<BString, Object> descriptor = ValueCreator.createMapValue();
+        descriptor.put(PD_VERSION, propertyDescriptor.version);
+        descriptor.put(PD_COPY_OPTIONS, propertyDescriptor.copyOptions);
+        descriptor.put(PD_OPTIONS, propertyDescriptor.options);
+        descriptor.put(PD_SUPPORT, propertyDescriptor.support);
+        descriptor.put(PD_CONTEXT, propertyDescriptor.context);
+        return descriptor;
+    }
+
+    public static BError createError(String errorType, String message, Throwable throwable) {
+        BError cause = ErrorCreator.createError(throwable);
+        BMap<BString, Object> errorDetails = ValueCreator.createRecordValue(getModule(), ERROR_DETAILS);
+        if (throwable instanceof MQException) {
+            errorDetails.put(ERROR_REASON_CODE, ((MQException) throwable).getReason());
+            errorDetails.put(ERROR_ERROR_CODE, ((MQException) throwable).getErrorCode());
+            errorDetails.put(ERROR_COMPLETION_CODE, ((MQException) throwable).getCompCode());
+        }
```

Comment:
As we are on java17, we don't need explicit castings
```suggestion
        if (throwable instanceof MQException exception) {
            errorDetails.put(ERROR_REASON_CODE, (exception.getReason());
            errorDetails.put(ERROR_ERROR_CODE, (exception.getErrorCode());
            errorDetails.put(ERROR_COMPLETION_CODE, (exception.getCompCode());
        }
```